azure/ConsiderationsForServiceDesign.md

```diff
@@ -116,6 +116,8 @@ For example, `collectedItems` not `itemsCollected`
 
 For example, `nextUrl` not `nextURL`.
 
+:ballot_box_with_check: **YOU SHOULD** use "DateTime" suffix in names of `date-time` values.
+
```

Comment:
Personally, I prefer At suffix. But I can live with DateTime

Reply:
If we have a consistent naming convention at the REST API level, we can choose to impose a different naming convention at the SDK level. So if you want to use the "At" suffix in Go SDKs -- if you think that would be idiomatic for Go -- then the generator can just make that substitution.

Reply:
I doubt a "DateTime" suffix would pass muster in any SDKs, right? With a lot of models in a response, using "At" or "On" will decrease the payload size.

Personally, I like "On" better. It works in more places than "At" e.g., "createdOn", "modifiedOn" or even "lastModifiedOn", "deletedOn", etc. This is what @KrzysztofCwalina had us use in track 2 wave 1 clients like Key Vault e.g., `CreatedOn`.

Reply:
I'd also prefer `On` or `At` to `DateTime`.  We're not putting any other type as a naming suffix so I don't see a benefit here.  Even if we wanted to make the argument it's because these are sent as strings over the wire, we don't use `ageInt` for formatting the values of numeric types, etc.

Reply:
+1 for `On` or `At` suffix. I think we should do a cursory check on existing datetime fields in the API specs to confirm it makes semantic sense for a broad variety of cases, but besides the payload advantages I like it better than `DateTime` type suffix which would be one-off compared to other types.

Reply:
This page has an interesting take on "on vs at":

https://www.differencebetween.com/difference-between-on-and-at/

Based on this, I think I would prefer the "At" suffix for `format: date-time` values.

Should we recommend "On" for `format: date` values?

Reply:
But our timestamps do include time, hence - even by that site's definition - "on" makes more sense. When, for example, have you heard someone say "modified at" or "deleted at"? According to Websters,

"On" _used as a function word to indicate a time frame during which something takes place_
"At" _used as a function word to indicate age or position in time_

I think switching between "on" vs. "at", though, is confusing. The nuance of one vs the other I imagine would be lost on many developers.

Reply:
I agree "At" is probably the best choice. But, as mentioned by @DominikMe ([here](https://github.com/microsoft/api-guidelines/pull/412#issuecomment-1386262617)), there are cases where the semantics benefits from terms like "From" or "To". Are these cases covered elsewhere? If not, there should be a note here.

Reply:
We don't cover cases that might benefit from "From" or "To".  The guidance for "At" is "YOU SHOULD" rather than "DO", so that allows other choices when appropriate. If you think we need something more specific, what would you propose?